### ISSUE TITLE
custom loss

### ISSUE THREAD

**Issue by wedesoft** [wedesoft/ppo#7]
```Python
import torch
import torch.nn as nn

class CustomMSELoss(nn.Module):
    def __init__(self):
        super(CustomMSELoss, self).__init__()

    def forward(self, predictions, targets):
        # Calculate the squared differences
        squared_diff = (predictions - targets) ** 2
        
        # Return the mean of those differences
        return torch.mean(squared_diff)

# --- Example Usage ---
# Simulated data
preds = torch.tensor([1.5, 2.5, 3.5], requires_grad=True)
labels = torch.tensor([1.0, 3.0, 4.0])

# Initialize and compute loss
criterion = CustomMSELoss()
loss = criterion(preds, labels)

print(f"Loss: {loss.item():.4f}")

# Backward pass to check if gradients flow
loss.backward()
print(f"Gradients on preds: {preds.grad}")
```
